graphql_compiler/compiler/emit_sql.py

```diff
@@ -125,10 +157,8 @@ def _find_folded_fields(ir):
         # Check the left and right side of Filter predicates to see if _x_count is involved.
         if global_operation_start and isinstance(block, blocks.Filter):
             for subexpression in (block.predicate.left, block.predicate.right):
-                if (isinstance(subexpression, FoldedContextField) and
-                        subexpression.fold_scope_location.field == COUNT_META_FIELD_NAME):
-                    fold_scope_location = subexpression.fold_scope_location
-                    fold_path = fold_scope_location.fold_path
+                fold_scope_location, fold_path = _find_x_count_used_in_global_filter(subexpression)
```

Comment:
I don't think I originally reviewed this code, so please pardon my confusion here. I'm just not sure I understand why we seem to be inferring (from the lowered IR) which folded fields are present where, when this information is already available from the query metadata and must have remained unchanged throughout lowering (or else the semantics of the query were altered by the lowering process and we have a bug in the system).

Reply:
good point! i will switch to using the output metadata table from the IR.